Update Receiver README (#234)
open-telemetry/opentelemetry-service: receiver/README.md

- [Zipkin Receiver](#zipkin)

## Configuring Receiver(s)

Receivers are configured via YAML under the top-level `receivers` tag. There
must be at least one enabled receiver for this configuration to be considered
valid.

The following is a sample configuration for the `examplereceiver`.
```yaml
receivers:
  # Receiver 1.
  # <receiver type>:
  examplereceiver:
    # <setting one>: <value one>
    endpoint: 1.2.3.4:8080
    # ...
  # Receiver 2.
  # <receiver type>/<name>:
  examplereceiver/settings:
    # <setting one>: <value one>
    disabled: false
    # <setting two>: <value two>
    endpoint: 127.0.0.1:9211
```

A receiver instance is referenced by its full name in other parts of the config,

> such as in pipelines. A full name consists of the receiver type, '/' and the

Reviewer: IMO naming convention can be a suggestion but users are free to define any names for receivers as long as they are unique.

Reviewer 2: @songy23 the name must start with receiver type otherwise it will not be recognised as a valid name. It be followed by a slash and a suffix.

name appended to the receiver type in the configuration. All receiver full names
must be unique. For the example above:
- Receiver 1 has full name `examplereceiver`.
- Receiver 2 has full name `examplereceiver/settings`.

> Most receiver expose a setting to disable it, by default receivers are enabled.

Reviewer: ```suggestion
Most receivers expose a setting to disable it, by default receivers are enabled.
```

Author: Fixed in follow up pr.

> Most receiver expose a setting to disable it, by default receivers are enabled.

Reviewer 2: AFAIK:
```suggestion
All receivers expose a setting to disable it, by default receivers are enabled.
```

Author: Fixed in follow up pr.

> At least one receiver has to be enabled for this part of the configuration to be

Reviewer 3: nit:  simpler sentance, if possible with a link to a pipeline descrition (TODO?)
```suggestion
At least one receiver is enabled in a valid pipeline configuration.
```

Author: Changed it to: `At least one receiver must be enabled to be a valid configuration.` yay/nay?

valid.

## <a name="opencensus"></a>OpenCensus Receiver
**Traces and metrics are supported.**

This receiver receives trace and metrics from [OpenCensus](https://opencensus.io/)
instrumented applications. It translates them into the internal format sent to
processors and exporters in the pipeline.

> To get started, all that is required is enable the OpenCensus receiver is to

Reviewer 3: ```suggestion
To get started, all that is required to enable the OpenCensus receiver is to
```

include it in the receiver definitions. This will enable the default values as
specified [here](https://github.com/open-telemetry/opentelemetry-service/blob/71589202609d7e787244076b631b45e219101867/receiver/opencensusreceiver/factory.go#L47-L56).
The following is an example:
```yaml
receivers:
  opencensus:
```

> The full list of settings exposed for this receiver are documented [here](https://github.com/open-telemetry/opentelemetry-service/blob/fdb5047ba75240fa3c0849474be97a3dbe213de8/receiver/opencensusreceiver/config.go#L28-L43)

Reviewer 2: I am not sure it is a good idea to reference a specific commit. If the code gets modified this will point to irrelevant information and we don't have a good way of knowing it went stale. Perhaps simply link to the latest config.go without using specific line numbers? 
Thoughts?

Author: I agree with the risk of it getting out of date. The benefit to adding a sha with specific lines is that we know it will directly link to the content we want users to see.

Since we seem to keep all of the config data in the config.go file and keep it strictly that right now there doesn't seem to be any down sides to just pointing to that file. If they get very large, then we can revisit to using a sha + line number.
